File: Roadmap/01 - OPERADORES Y ESTRUCTURAS DE CONTROL/javascript/Roadmap/00 - SINTAXIS, VARIABLES, TIPOS DE DATOS Y HOLA MUNDO/c/othamae.c
/*
 * - Crea un comentario en el código y coloca la URL del sitio web oficial del
 *   lenguaje de programación que has seleccionado.
 * - Representa las diferentes sintaxis que existen de crear comentarios
 *   en el lenguaje (en una línea, varias...).
 * - Crea una variable (y una constante si el lenguaje lo soporta).
 * - Crea variables representando todos los tipos de datos primitivos
 *   del lenguaje (cadenas de texto, enteros, booleanos...).
 * - Imprime por terminal el texto: "¡Hola, [y el nombre de tu lenguaje]!"
 *
 * ¿Fácil? No te preocupes, recuerda que esta es una ruta de estudio y
 * debemos comenzar por el principio.
 */


// https://www.gnu.org/software/gnu-c-manual/gnu-c-manual.html
// https://en.wikibooks.org/wiki/C_Programming
// https://devdocs.io/c/

// Comment in one line

/**
 * comment
 * in
 * block
 */

#include <stdio.h>

#define CONSTANT "This is a constant"


int main() {

    // Variables
    char letter = 'a';
    int integer = 1;
    float floating = 10.5;
    double double_floating = -1.35;
        
    printf("Hello C!\n");
    printf("This is a constant: %s\n", CONSTANT);
    printf("This is a char: %c\n", letter);
    printf("This is an integer: %d\n", integer);
    printf("This is a float: %g\n", floating);
    printf("This is a double: %lf\n", double_floating);
    return 0;
}

File: Roadmap/00 - SINTAXIS, VARIABLES, TIPOS DE DATOS Y HOLA MUNDO/javascript/othamae.js
/*
 * - Crea un comentario en el código y coloca la URL del sitio web oficial del
 *   lenguaje de programación que has seleccionado.
 * - Representa las diferentes sintaxis que existen de crear comentarios
 *   en el lenguaje (en una línea, varias...).
 * - Crea una variable (y una constante si el lenguaje lo soporta).
 * - Crea variables representando todos los tipos de datos primitivos
 *   del lenguaje (cadenas de texto, enteros, booleanos...).
 * - Imprime por terminal el texto: "¡Hola, [y el nombre de tu lenguaje]!"
 *
 * ¿Fácil? No te preocupes, recuerda que esta es una ruta de estudio y
 * debemos comenzar por el principio.
 */


// https://developer.mozilla.org/en-US/docs/Web/JavaScript

// Comment in one line

/**
 * comment
 * in
 * block
 */


const constant = 'This is a constant'

let variable = 'This is a variable'

var globalVariable= 'This is a global variable'


// 7 PRIMITIVE TYPES:

// 1 - String
let string = 'This is a string'

// 2 - Number   
let number = 10
let float = 10.5

// 3 - BigInt
let bigInt = BigInt(9007199254740991) // 9007199254740991n

// 4 - Boolean
let boolean = true

// 5 - Undefined
let undefinedVariable

// 6 - Symbol
let symbol = Symbol('This is a symbol')

// 7 - Null
let nullVariable = null


// Print in terminal:
console.log("¡Hola, JavaScript!")